src/main/java/ch/admin/bag/covidcertificate/service/CleanupService.java

```diff
@@ -0,0 +1,154 @@
+package ch.admin.bag.covidcertificate.service;
+
+import ch.admin.bag.covidcertificate.config.cleanup.Cleanup;
+import ch.admin.bag.covidcertificate.config.cleanup.Database;
+import ch.admin.bag.covidcertificate.config.cleanup.SqlQuery;
+import com.zaxxer.hikari.HikariDataSource;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDate;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class CleanupService {
+
+    /**
+     * Deletes the next portion of records limited by given date.<br>
+     * The number of records to delete may be limited to batch-size by the <code>asBatch</code> flag.
+     *
+     * @param effort  the database to delete a batch from
+     * @param date    records older than this date are delete-candidates
+     * @param asBatch <code>true</code> demands deletion of configured batch-size,
+     *                <code>false</code> deletes everything
+     * @return number of records this batch deleted
+     */
+    @Transactional
+    public long delete(CleaningEffort effort, LocalDate date, boolean asBatch) {
+        long count = -1L;
+        if (effort != null) {
+            if (asBatch) {
+                count = effort.jdbcTemplate.update(effort.deleteUntilBatchQuery, new Object[] { date, effort.deleteUntilBatchSize});
+            } else {
+                count = effort.jdbcTemplate.update(effort.deleteUntilQuery, new Object[] { date });
+            }
+        }
+        return count;
+    }
+
+    public long getCount(CleaningEffort effort, LocalDate date) {
+        long count = -1L;
+        if (effort != null) {
+            count = effort.jdbcTemplate.queryForObject(effort.countQuery, Long.class, new Object[] { date } );
+        }
+        return count;
+    }
+
+    public CleaningEffort create(String databaseName, Cleanup cleanup) {
+
+        CleaningEffort effort = null;
+
+        boolean nameValid = StringUtils.isNoneBlank(databaseName);
+        if (!nameValid) {
+            log.debug("CLEANING | Init {} - name '{}' invalid", databaseName, databaseName);
+        }
+
+        Database db = cleanup.getDatabase();
+        boolean databaseValid = false;
+        if (db == null) {
+            log.debug("CLEANING | Init {} - database config for '{}' missing", databaseName, databaseName);
+        } else {
+            databaseValid = StringUtils.isNoneBlank(db.getDriverClassName(), db.getUrl(), db.getUsername(), db.getPassword());
+            if (!databaseValid) {
+                log.debug("CLEANING | Init {} - database config invalid:\n driverClassName='{}'\n url='{}'\n username='{}'\n password='***",
```

Comment:
I think we should remove the password text here as it has no information for any reader of the log.